Add adr 0003 Default PHP Extensions (#77)
usabilla/php-docker-template: docs/adr/0003-default-php-extensions.md

# 3. Default PHP Extensions

Date: 2019-02-27

## Status

Accepted

## Context

Given the premise those images are designed to run in production, in a clustered environment and managed by an orchestrator like Kubernetes or Docker Swarm, it becomes necessary to shape the PHP installation towards those needs.

For instance by default PHP code won't be able to handle `SIGTERM` or `SIGINT` from the orchestrator, in memory user cache won't come by default, which can be useful for projecting data into a service, and more.

## Decision

Ship the Docker images with extensions which contribute to the vision of the microservices use case, being:

- `PCNTL` in order to deal with user signals. [PCNTL Manual page](http://php.net/manual/en/book.pcntl.php)

>   - `fpm` won't be impacted since it deals with the signals, this is meant for the long running non-interactive php applications

Reviewer: ```suggestion
  - This is meant for long running non-interactive php applications. `fpm` is not impacted, since it can deal with these signals. 
```
